src/platform/baytrail/include/platform/lib/memory.h

```diff
@@ -24,7 +24,7 @@ void platform_init_memmap(void);
 #define SHIM_SIZE	0x00004000
 
 #define IRAM_BASE	0xFF2C0000
-#define IRAM_SIZE	0x00014000
+#define IRAM_SIZE	0x00015000
```

Comment:
IRAM size is limited by hardware. I don't think that BYT has over 0x14000 bytes.

Reply:
xcc build for CHT have the final result.
```
        Totals  Start           End             Size
        TEXT    0xff2c0000      0xff2d405a      0x1405a
        DATA    0xff300010      0xff328000      0x27ff0
        BSS     0xff303970      0xff3075d0      0x3c60
```
gcc build
```
        Totals  Start           End             Size
        TEXT    0xff2c0000      0xff2d244a      0x1244a
        DATA    0xff300008      0xff3071c4      0x71bc
        BSS     0xff303510      0xff307158      0x3c48
```
As I can see we have more driver for CHT then BYT.
Is CHT have same limitation with BYT?

https://github.com/thesofproject/sof/blob/fe321d65f8f1034a31d81289599dc281f3d44ef1/src/platform/baytrail/baytrail.x.in#L81-L83

Reply:
@xiulipan 
Can you try changing `-O2` to `-Os` in  line:
```
target_compile_options(sof_options INTERFACE $<$<COMPILE_LANGUAGE:C>:-O2 -g -Wall -Werror -Wl,-EL -Wmissing-prototypes -mtext-section-literals>) 
```
In `src/arch/xtensa/CMakeLists.txt` ?

For me on APL it reduced size from 0x209da to 0x1d1ba.

Reply:
@jajanusz It help!
```
writing module 0 sof-cht

        Totals  Start           End             Size
        TEXT    0xff2c0000      0xff2d22a2      0x122a2
        DATA    0xff300010      0xff328000      0x27ff0
        BSS     0xff303730      0xff307390      0x3c60
```

Will this influence other platform?

And I see some issues from 5c7a49af38dc6129890e37c51f2a2bd57f0c4134 to cause hifi2 build error

```
/home/pxl/work/sof/sof/src/audio/src/src_hifi2ep.c: In function ‘src_polyphase_stage_cir’:
/home/pxl/work/sof/sof/src/audio/src/src_hifi2ep.c:384: warning: pointer of type ‘void *’ used in arithmetic
/home/pxl/work/sof/sof/src/audio/src/src_hifi2ep.c: In function ‘src_polyphase_stage_cir_s16’:
/home/pxl/work/sof/sof/src/audio/src/src_hifi2ep.c:510: warning: pointer of type ‘void *’ used in arithmetic
```

Will send fix for this also.